Plugins/Public/balancemagic/Main.cpp

```diff
@@ -345,296 +198,69 @@ bool UserCmd_Process(uint iClientID, const wstring &wscCmd)
+			float curr, max;
 			bool bShieldsUp = false;
 
-			if (p1 == 1)
-				pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
-			else if (p1 == 65521)
+			if (subObjID == 1) // 1 is base (hull)
+				pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max); 
+			else if (subObjID == 65521) // 65521 is shield (bubble, not equipment)
 				pub::SpaceObj::GetShieldHealth(iDmgToSpaceID, curr, max, bShieldsUp);
+			else
+				return; // If hit mounted equipment - do not continue with uninitialized variables.
 
-			/*PrintUserCmdText(iDmgFrom, L"HkCb_AddDmgEntry CORONA iDmgToSpaceID=%u get_inflictor_id=%u curr=%0.2f max=%0.0f damage=%0.2f cause=%u is_player=%u player_id=%u fate=%u\n",
-			iDmgToSpaceID, dmg->get_inflictor_id(), curr, max, damage, dmg->get_cause(), dmg->is_inflictor_a_player(), dmg->get_inflictor_owner_player(), fate);*/
-
-			if (targetShipClass == 0 || targetShipClass == 1 || targetShipClass == 3)
-				adjustedDamage = curr - iter->second.fighter / (bShieldsUp ? 2 : fTargetArmorScale);
-			else if (targetShipClass == 2 || targetShipClass == 4 || targetShipClass == 5 || targetShipClass == 19)
-				adjustedDamage = curr - iter->second.freighter / (bShieldsUp ? 2 : fTargetArmorScale);
-			else if (targetShipClass < 11)
-				adjustedDamage = curr - iter->second.transport / (bShieldsUp ? 2 : fTargetArmorScale);
-			else if (targetShipClass < 13)
-				adjustedDamage = curr - iter->second.gunboat / (bShieldsUp ? 2 : fTargetArmorScale);
-			else if (targetShipClass < 15)
-				adjustedDamage = curr - iter->second.cruiser / (bShieldsUp ? 2 : fTargetArmorScale);
-			else if (targetShipClass < 16)
-				adjustedDamage = curr - iter->second.battlecruiser / (bShieldsUp ? 2 : fTargetArmorScale);
-			else if (targetShipClass < 19)
-				adjustedDamage = curr - iter->second.battleship / (bShieldsUp ? 2 : fTargetArmorScale);
-
-			dmg->add_damage_entry(p1, adjustedDamage, fate);
-
-			if (set_iPluginDebug)
-				PrintUserCmdText(iDmgFrom, L"You hit a ship's hull (ID 0x%08X) with an 0x%08X for %f adjusted damage (%f -> %f) (p1 = %u, fate = %u).", uArchID, iDmgMunitionID, (curr - adjustedDamage), curr, adjustedDamage, p1, fate);
-
-			iDmgTo = 0;
-			iDmgToSpaceID = 0;
-			iDmgMunitionID = 0;
+			uint iTargetType;
+			pub::SpaceObj::GetType(iDmgToSpaceID, iTargetType);
 
-			returncode = SKIPPLUGINS_NOFUNCTIONCALL;
-		}
-		else
-			return;
-	}
-}
-
-void __stdcall FireWeapon(unsigned int iClientID, struct XFireWeaponInfo const &wpn)
-{
-	returncode = DEFAULT_RETURNCODE;
-
-	if (!iLoadedBurstFires)
-		return;
-
-	IObjInspectImpl *obj = HkGetInspect(iClientID);
-	if (obj)
-	{
-		CShip* cship = (CShip*)HkGetEqObjFromObjRW((IObjRW*)obj);
-
-		for (short* slot = wpn.sArray1; slot != wpn.sArray2; slot++)
-		{
-			CEquipManager* eqmanager = (CEquipManager*)((char*)cship + 0xE4);
-			CEquip *equip = eqmanager->FindByID(*slot);
-			if (CEGun::cast(equip))
+			// Deduce: if not fighter nor freighter, then it's obviously solar object.
```

Comment:
this sounds dangerous - there are lots of other types of space objects defined in FLCoreCommon.h

Reply:
We discussed this and it's fine. FL only lets us have fighters, freighters, and solars. And probably loot in space but we don't care about that.